src/osemgrep/reporting/Cli_json_output.mli

```diff
@@ -14,5 +14,12 @@ val exit_code_of_error_type : Semgrep_output_v1_t.core_error_kind -> Exit_code.t
 val cli_match_of_core_match :
   Rule.hrules -> Semgrep_output_v1_t.core_match -> Semgrep_output_v1_t.cli_match
 
+val index_match_based_ids :
+  Semgrep_output_v1_t.cli_match list -> Semgrep_output_v1_t.cli_match list
+(** [index_match_based_ids matches] will append an index to the match based id
```

Comment:
but the match based id is a string right? You mean an index is appended to the string?
Isn't the match_based_id supposed to be a match identifier ... why we need this extra index?

Reply:
I'm adding an explanation, but the tldr is we do some logic on the app side for deduplication based on the index of the id.